test/unit/math/mix/fun/cholesky_factor_constrain_test.cpp

```diff
@@ -22,14 +23,14 @@ TEST(mathMixMatFun, cholesky_factor_constrain_lp) {
   auto f1 = [](int M, int N) {
     return [M, N](const auto& x1) {
       stan::scalar_type_t<std::decay_t<decltype(x1)>> lp = 0.0;
-      return stan::math::cholesky_factor_constrain(x1, M, N, lp);
+      return stan::math::cholesky_factor_constrain<true>(x1, M, N, lp);
     };
   };
 
   auto f2 = [](int M, int N) {
     return [M, N](const auto& x1) {
       stan::scalar_type_t<std::decay_t<decltype(x1)>> lp = 0.0;
-      stan::math::cholesky_factor_constrain(x1, M, N, lp);
+      stan::math::cholesky_factor_constrain<true>(x1, M, N, lp);
```

Comment:
[question]
Are we still testing `cholesky_factor_constrain`?  Given that it's still in the API, it should still be tested.  Especially since I've filed the issue to refactor the implementations. 

Same question for all of these tests.

Reply:
Yep! See my comment [here](https://github.com/stan-dev/math/pull/2559#discussion_r694482314)